batching/batching.go

```diff
@@ -208,7 +235,7 @@ func (b *Batcher) run() {
 	for {
 		timer.Reset(b.timespan)
 
```

Comment:
I don't think we should use the passed context.Context here directly since we derive it from New(). 

Context usually is scoped by a request lifetime, this is tracking New()'s lifetime.  What we really are doing is getting something for metrics that is being passed inside Context.

Not that I expect someone to cancel on us, but we can do: `b.handleInput(context.WithoutCancel(ctx), timer.C)` which is really rescoping for this call.

Reply:
Makes sense, I just wanted to make sure the context was being passed down to the metric recording and went a little too far.

Reply:
I'm creating the new context in handleInput.